ajax() type signature error (TS) when string provided

John Lindquist reported that

```ts
import {Observable} from 'rxjs/Rx.DOM';
Observable.ajax('http://swapi.co/api/people/1/')
    .subscribe();
```

gives TS compile error
```
ERROR in ./src/main.ts
(2,1): error TS2346: Supplied parameters do not match any signature of call target.
```

---

Solution suggestion:
I think the type signature should be changed like this:

```diff
 export interface AjaxCreationMethod {
-  (): <T>(urlOrRequest: string | AjaxRequest) => Observable<T>;
+  <T>(urlOrRequest: string | AjaxRequest): Observable<T>;
   get: <T>(url: string, resultSelector?: (response: AjaxResponse) => T, headers?: Object) => Observable<T>;
   post: <T>(url: string, body?: any, headers?: Object) => Observable<T>;
   put: <T>(url: string, body?: any, headers?: Object) => Observable<T>;
   delete: <T>(url: string, headers?: Object) => Observable<T>;
   getJSON: <T, R>(url: string, resultSelector?: (data: T) => R, headers?: Object) => Observable<R>;
 }
```

--- a/src/observable/dom/AjaxObservable.ts
+++ b/src/observable/dom/AjaxObservable.ts
@@ -39,12 +39,12 @@ const createXHRDefault = (): XMLHttpRequest => {
 };
 
 export interface AjaxCreationMethod {
-  (): <T>(urlOrRequest: string | AjaxRequest) => Observable<T>;
-  get: <T>(url: string, resultSelector?: (response: AjaxResponse) => T, headers?: Object) => Observable<T>;
-  post: <T>(url: string, body?: any, headers?: Object) => Observable<T>;
-  put: <T>(url: string, body?: any, headers?: Object) => Observable<T>;
-  delete: <T>(url: string, headers?: Object) => Observable<T>;
-  getJSON: <T, R>(url: string, resultSelector?: (data: T) => R, headers?: Object) => Observable<R>;
+  <T>(urlOrRequest: string | AjaxRequest): Observable<T>;
+  get<T>(url: string, resultSelector?: (response: AjaxResponse) => T, headers?: Object): Observable<T>;
+  post<T>(url: string, body?: any, headers?: Object): Observable<T>;
+  put<T>(url: string, body?: any, headers?: Object): Observable<T>;
+  delete<T>(url: string, headers?: Object): Observable<T>;
+  getJSON<T, R>(url: string, resultSelector?: (data: T) => R, headers?: Object): Observable<R>;
 }
 
 function defaultGetResultSelector<T>(response: AjaxResponse): T {

